com.unity.multiplayer.mlapi/Tests/Runtime/NetworkUpdateLoopTests.cs

```diff
@@ -11,8 +11,45 @@ namespace MLAPI.RuntimeTests
 {
     public class NetworkUpdateLoopTests
     {
+        [UnityTest]
+        public IEnumerator RegisterAndUnregisterSystems()
+        {
+            // caching the current PlayerLoop (it will have NetworkUpdateLoop systems registered)
+            var cachedPlayerLoop = PlayerLoop.GetCurrentPlayerLoop();
+            {
+                // since current PlayerLoop already took NetworkUpdateLoop systems inside,
+                // we are going to swap it with the default PlayerLoop temporarily for testing
+                PlayerLoop.SetPlayerLoop(PlayerLoop.GetDefaultPlayerLoop());
+                var oldPlayerLoop = PlayerLoop.GetCurrentPlayerLoop();
+
+                NetworkUpdateLoop.RegisterLoopSystems();
+
+                int waitFrameNumber = Time.frameCount + 8;
```

Comment:
What's the purpose of this? Why 8 frames instead of 1? Is it "in case some other system registers itself"? I hate timing-based tests and would love to help you remove this aspect of the test if possible.

Reply:
yeah — no particular reason for multiple frames, I'll make it the next frame only (`Time.frameCount + 1`).